datafusion/common/src/scalar.rs

```diff
@@ -1562,7 +1597,11 @@ impl ScalarValue {
                     DataType::UInt16 => dict_from_values::<UInt16Type>(values)?,
                     DataType::UInt32 => dict_from_values::<UInt32Type>(values)?,
                     DataType::UInt64 => dict_from_values::<UInt64Type>(values)?,
-                    _ => unreachable!("Invalid dictionary keys type: {:?}", key_type),
+                    _ => {
+                        return _internal_err!(
+                            "Invalid dictionary keys type: {key_type:?}"
+                        )
+                    }
```

Comment:
I believe this error is reachable if the user passes dictionaries with non-integer keys to the function. It is recoverable so I replaced `unreachable!` with an internal error.